plumbing/object/commit_walker_bfs.go

```diff
@@ -64,6 +64,21 @@ func (w *bfsCommitIterator) Next() (*Commit, error) {
 
 		w.seen[c.Hash] = true
 
+		if storer, ok := c.s.(storer.ShallowStorer); ok {
+			shallowCommits, err := storer.Shallow()
```

Comment:
Is this something we would only want to do once and reuse within the `for` loop?

Reply:
I was looking at that. The only issue is that `c.s` is only available in the loop context, so unless there is another way to obtain the storer, we would need to add logic to check if we already obtained the shallow commits.

Reply:
And it turns out that once it reaches that part of the loop, it will always return from the function, so it will only execute once anyway.

Reply:
Ahh, fair enough, I wonder if we should change the logic fractionally to close the for loop before that point, or at least remove the `var c` outside the loop as its useless as is.